tests/Content/TranslationTest.php

```diff
@@ -0,0 +1,361 @@
+	}
+
+	/**
+	 * @covers ::exists
+	 */
+	public function testExistsMultiLanguage()
+	{
+		$this->setUpMultiLanguage();
+
+		$translationEN = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('en')
+		);
+
+		$translationDE = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('de')
+		);
+
+		$this->assertFalse($translationEN->version()->exists());
+		$this->assertFalse($translationDE->version()->exists());
+
+		$this->createContentMultiLanguage();
+
+		$this->assertTrue($translationEN->version()->exists());
+		$this->assertTrue($translationDE->version()->exists());
+	}
+
+	/**
+	 * @covers ::exists
+	 */
+	public function testExistsSingleLanguage()
+	{
+		$this->setUpSingleLanguage();
+
+		$translation = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::single()
+		);
+
+		$this->assertFalse($translation->version()->exists());
+
+		$this->createContentSingleLanguage();
+
+		$this->assertTrue($translation->version()->exists());
+	}
+
+	/**
+	 * @covers ::isDefault
+	 */
+	public function testIsDefault()
+	{
+		$this->setUpMultiLanguage();
+
+		$en = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('en')
+		);
+
+		$de = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('de')
+		);
+
+		$this->assertTrue($en->language()->isDefault());
+		$this->assertFalse($de->language()->isDefault());
+	}
+
+	/**
+	 * @covers ::language
+	 */
+	public function testLanguage()
+	{
+		$this->setUpMultiLanguage();
+
+		$translationEN = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: $languageEN = Language::ensure('en')
+		);
+
+		$translationDE = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: $languageDE = Language::ensure('de')
+		);
+
+		$this->assertSame($languageEN, $translationEN->language());
+		$this->assertSame($languageDE, $translationDE->language());
+	}
+
+	/**
+	 * @covers ::model
+	 */
+	public function testModel()
+	{
+		$this->setUpMultiLanguage();
+
+		$translation = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('en')
+		);
+
+		$this->assertSame($this->model, $translation->model());
+	}
+
+	/**
+	 * @covers ::slug
+	 */
+	public function testSlugExists()
+	{
+		$this->setUpMultiLanguage();
+
+		$translationEN = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('en')
+		);
+
+		$translationDE = new Translation(
+			model: $this->model,
+			version: $this->model->version(),
+			language: Language::ensure('de')
+		);
+
+		Data::write($this->contentFile('en'), [
+			'title' => 'Test',
+			'slug'  => 'english-slug'
```

Comment:
`Todo`: Getting the slug of a non-default language is not something we do in the core if I remember correctly as we dropped support for storing the slug field in the default language content for performance reasons. So I wouldn't test it either (allows us to change that later without breaking the test that tests what we don't actually need).